components/tools/OmeroJava/test/integration/LightAdminRolesTest.java

```diff
@@ -0,0 +1,2222 @@
+        for (final String permission : permissions) {
+            final AdminPrivilege privilege = new AdminPrivilegeI();
+            privilege.setValue(omero.rtypes.rstring(permission));
+            privileges.add(privilege);
+        }
+        rootSession.getAdminService().setAdminPrivileges(user, privileges);
+        /* avoid old session as privileges are briefly cached */
+        loginUser(ctx);
+        return ctx;
+    }
+
+    /**
+     * Sudo to the given user.
+     * @param target a user
+     * @return context for a session owned by the given user
+     * @throws Exception if the sudo could not be performed
+     */
+    private EventContext sudo(Experimenter target) throws Exception {
+        if (!target.isLoaded()) {
+            target = iAdmin.getExperimenter(target.getId().getValue());
+        }
+        final Principal principal = new Principal();
+        principal.name = target.getOmeName().getValue();
+        final Session session = factory.getSessionService().createSessionWithTimeout(principal, 100 * 1000);
+        final omero.client client = newOmeroClient();
+        final String sessionUUID = session.getUuid().getValue();
+        client.createSession(sessionUUID, sessionUUID);
+        return init(client);
+    }
+
+    /**
+     * Test that an ImporterAs can create new Project and Dataset
+     * and import data on behalf of another user solely with <tt>Sudo</tt> privilege
+     * into this Dataset. Further link the Dataset to the Project, check
+     * that the link belongs to the user (not to the ImporterAs).
+     * All workflows are tested here both when light admin is sudoing
+     * and when he/she is not sudoing, except for Link and Import (both tested
+     * only when sudoing, as the non-sudoing workflows are too complicated
+     * for those two actions and thus covered by separate tests.
+     * @param isAdmin if to test a member of the <tt>system</tt> group
+     * @param isSudoing if to test a success of workflows where Sudoed in
+     * @param permWriteOwned if to test a user who has the <tt>WriteOwned</tt> privilege
+     * @param groupPermissions if to test the effect of group permission level
+     * @throws Exception unexpected
+     */
+    @Test(dataProvider = "isSudoing and WriteOwned privileges cases")
+    public void testImporterAsSudoCreateImport(boolean isSudoing, boolean permWriteOwned,
+            String groupPermissions) throws Exception {
+        final EventContext normalUser = newUserAndGroup(groupPermissions);
+        final boolean isExpectSuccessCreate = permWriteOwned || isSudoing;
+        /* set up the light admin's permissions for this test */
+        List<String> permissions = new ArrayList<String>();
+        permissions.add(AdminPrivilegeSudo.value);
+        if (permWriteOwned) permissions.add(AdminPrivilegeWriteOwned.value);
+        final EventContext lightAdmin;
+        lightAdmin = loginNewAdmin(true, permissions);
+        if (isSudoing) {
+            try {
```

Comment:
Will fix at a later stage, captured in first checklist point of https://trello.com/c/ygDN23CC/41-new-role-integration-tests-petr